### PR TITLE
SOLR-11657: Added @Deprecated annotation

### PR DESCRIPTION
**Pull request by joshgog**
https://issues.apache.org/jira/browse/SOLR-11657

# Description

Deprecate ContentStream out of SolrJ

# Solution

Annotated the class with @Deprecated annotation

# Tests

Not required.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by epugh** [comment]
Can you reformat this to indicate what people *should* use instead?    I searched for `@deprecated` and found a bunch of examples in the source like.   See how nice the Javadocs are to cue you where to go?

```
  /**
   * Get baseUrl as StringBuffer for back compat with previous version
   *
   * @deprecated Please use {@link #getBaseUrl()} instead
   * @return StringBuffer of the base url without query part
   */
  @Deprecated
```

In terms of marking this as Deprecated, it makes sense to me, however I hope @dsmiley weighs in as he commented on the JIRA.

**Comment by epugh** [comment]
@joshgog I added this to the changes file, how do you want to be credited?   

@dsmiley can you review?

**Comment by joshgog** [comment]
> @joshgog I added this to the changes file, how do you want to be credited?

I'm OK with` (Joshua Ouma via Eric Pugh)`

File: solr/solrj/src/java/org/apache/solr/common/util/ContentStream.java [hunk 1]
```diff
@@ -19,10 +19,13 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.Reader;
+import org.apache.solr.client.solrj.request.RequestWriter;
 
 /**
  * @since solr 1.2
+ * @deprecated Use {@link RequestWriter.ContentWriter} instead.
```
Comment (joshgog) [inline solr/solrj/src/java/org/apache/solr/common/util/ContentStream.java:26]: This causes the CI test to fail due to `Javadoc: Invalid member type qualification`. I think `@deprecated Use {@link ContentWriter} instead`. should work fine

**Review by risdenk — APPROVED** [review]

**Pull request merged**